applications/MetisApplication/tests/quad_partition_test.py

```diff
@@ -0,0 +1,46 @@
+import os
+import shutil
+
+from KratosMultiphysics import KratosUnittest, Model, ModelPartIO, PARTITION_INDEX, FRACTIONAL_STEP
+from KratosMultiphysics.MetisApplication import MetisDivideHeterogeneousInputProcess
+
+class QuadPartitionTest(KratosUnittest.TestCase):
+
+    def setUp(self):
+        self.work_dir = os.path.abspath(os.path.dirname(__file__))
+
+    def testConditionPartitioning(self):
+        self.test_file = test_file = "quads"
+        partitions = 3
+        sync_conditions = True
+        domain_size = 2
+        verbosity = 0
+
+        model = Model()
+        io_flags = ModelPartIO.READ|ModelPartIO.SKIP_TIMER|ModelPartIO.IGNORE_VARIABLES_ERROR
+
+        with KratosUnittest.WorkFolderScope(self.work_dir, ''):
+            io = ModelPartIO(test_file, io_flags)
+
+            partitioner = MetisDivideHeterogeneousInputProcess(
+                io, partitions, domain_size, verbosity, sync_conditions)
+            partitioner.Execute()
+
+            for i in range(partitions):
+                model_part = model.CreateModelPart(f'test_{i}')
+                model_part.AddNodalSolutionStepVariable(PARTITION_INDEX)
+
+                file_path = f'{test_file}_partitioned/{test_file}_{i}'
+                partitioned_io = ModelPartIO(file_path, io_flags)
+                partitioned_io.ReadModelPart(model_part)
+
+                for condition in model_part.Conditions:
+                    parent = condition.GetValue(FRACTIONAL_STEP)
+                    self.assertTrue(parent in model_part.Elements)
+
+    def tearDown(self):
+        with KratosUnittest.WorkFolderScope(self.work_dir, ''):
+            try:
+                shutil.rmtree(f'{self.test_file}_partitioned')
+            except:
+                pass
```

Comment:
this can be done better:
https://github.com/KratosMultiphysics/Kratos/blob/5a9b687fb858fc9c2bcd8c86c0eadd8d799c5d1a/applications/CoSimulationApplication/tests/co_simulation_test_factory.py#L183

Reply:
Updated to use the cleanUp method, but be careful with the linked example: if you register the cleanUp operation after running the test, they will not run if the test ever fails.